plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/PointStroke.tsx

```diff
@@ -1,9 +1,186 @@
+import { styled } from "@web/theme";
+import { useCallback, useState } from "react";
+
+import { BaseFieldProps, Cond, Expression, Fields } from "../types";
+
+import { array_move } from "./common/common";
+import ConditionField from "./common/ConditionField";
+import Field from "./common/Field";
+import ItemControls from "./common/ItemControls";
 
 const PointStroke: React.FC<BaseFieldProps<"pointStroke">> = ({ value, editMode, onUpdate }) => {
   // remember to update the BaseFieldProps type!
-  console.log(value, editMode, onUpdate);
-  return <div>PointColor</div>;
+  const [conditions, updateConditions] = useState(value.conditions);
+
+  const handleMoveUp = useCallback(
+    (idx: number) => {
+      if (idx === 0) return;
+      updateConditions(c => {
+        let newConditions: Fields["pointStroke"]["conditions"] = undefined;
+
+        if (c) {
+          newConditions = c;
+          array_move(newConditions, idx, idx - 1);
+        }
+        onUpdate({
+          ...value,
+          conditions: newConditions,
+        });
+        return newConditions;
+      });
+    },
+    [value, onUpdate],
+  );
+
+  const handleMoveDown = useCallback(
+    (idx: number) => {
+      if (conditions && idx >= conditions.length - 1) return;
+      updateConditions(c => {
+        let newConditions: Fields["pointStroke"]["conditions"] = undefined;
+        if (c) {
+          newConditions = c;
+          array_move(newConditions, idx, idx + 1);
+        }
+        onUpdate({
+          ...value,
+          conditions: newConditions,
+        });
+        return newConditions;
+      });
+    },
+    [conditions, onUpdate, value],
+  );
+
+  const handleAdd = useCallback(() => {
+    updateConditions(c => {
+      const newCondition: {
+        expression: Expression;
+        strokeColor: string;
+        strokeWidth: number;
+      } = {
+        expression: {
+          conditions: [
+            {
+              key: "ARGH",
+              operator: "=",
+              operand: "AField",
+              value: 1,
+            },
+          ],
+        },
+        strokeColor: "brown",
+        strokeWidth: 10,
+      };
+      onUpdate({
+        ...value,
+        conditions: value.conditions ? [...value.conditions, newCondition] : [newCondition],
+      });
+      return c ? [...c, newCondition] : [newCondition];
+    });
+  }, [value, onUpdate]);
+
+  const handleRemove = useCallback(
+    (idx: number) => {
+      updateConditions(c => {
+        let newConditions: Fields["pointStroke"]["conditions"] = undefined;
+        if (c) {
+          newConditions = c.filter((_, idx2) => idx2 != idx);
+        }
+        onUpdate({
+          ...value,
+          conditions: newConditions,
+        });
+        return newConditions;
+      });
+    },
+    [value, onUpdate],
+  );
+
+  // this is hard-codded condition
+  const condition: Cond<number> = {
+    key: "ARGH",
+    operator: "=",
+    operand: "AField",
+    value: 1,
+  };
```

Comment:
Is this needed outside of development? Since you are linked up to value and onUpdate, isn't actual user customizing possible?

Reply:
Done!